chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -30,16 +30,30 @@ object Mem {
   }
 }
 
-sealed abstract class MemBase[T <: Data](t: T, val length: Int) extends HasId with VecLike[T] {
+/* The following definition of ChiselVecLike is to work around the issue that you cannot
```

Comment:
I'm not sure I completely grok the suggestion. We need a concrete `apply(x: Int): T` method that the macro can override. I've added a suitable definition to `VecLike`. Is this what was intended?